app/Console/Commands/Indexer/MangaSweepIndexer.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace App\Console\Commands\Indexer;
+
+use App\Exceptions\Console\CommandAlreadyRunningException;
+use App\Exceptions\Console\FileNotFoundException;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Storage;
+
+/**
+ * Class MangaSweepIndexer
+ *
+ * @package App\Console\Commands\Indexer
+ */
+class MangaSweepIndexer extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *`
+     *
+     * @var string
+     */
+    protected $signature = 'indexer:manga-sweep';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Delete all removed manga';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     * @throws FileNotFoundException
+     */
+    public function handle()
+    {
+        $this->info("Info: Delete removed MAL IDs\n\n");
+
+        echo "Loading MAL IDs\n";
+        $malIds = array_fill_keys($this->fetchMalIds(), null);
+
+        echo "Loading MAL IDs from local DB\n";
+        $results = DB::table('manga')->select('mal_id', '_id')->get();
+
+        echo "Compare MAL IDs\n";
+        $remove = [];
+
+        foreach ($results as $result) {
+            if (!array_key_exists($result['mal_id'], $malIds)) {
+                $remove[] = $result['_id'];
+            }
+        }
+
+        echo "Delete removed MAL IDs\n";
+        DB::table('manga')->whereIn('_id', $remove)->delete();
+    }
+
+    /**
+     * @return array
+     * @url https://github.com/seanbreckenridge/mal-id-cache
+     */
+    private function fetchMalIds(): array
```

Comment:
this could go in a trait or a base class i guess

Reply:
@pushrbx @nerg4l any changes needed here for this PR?

Reply:
@irfan-dahir I will create an abstraction for fetching from mal-id-cache. In my opinion, this PR can be merged regardless.